src/SessionManager.h

```diff
@@ -141,6 +141,59 @@ class SessionManager final {
 
 private:
 
+	class StatBlocks {
+
+	public:
+
+		struct Block {
+			telemetry::IntGauge num;
+			telemetry::IntCounter total;
+			size_t max = 0;
+
+			Block(telemetry::IntGaugeFamily num_family,
+			      telemetry::IntCounterFamily total_family,
+			      std::string protocol) : num(num_family.GetOrAdd({{"protocol", protocol}})),
+			                              total(total_family.GetOrAdd({{"protocol", protocol}}))
+				{
+				}
+			};
+
+		using BlockMap = std::map<std::string, Block>;
+
+		BlockMap::iterator InitCounters(std::string protocol)
+			{
+			telemetry::IntGaugeFamily num_family = telemetry_mgr->GaugeFamily(
+				"zeek", "open-sessions", {"protocol"}, "Active Zeek Sessions");
+			telemetry::IntCounterFamily total_family = telemetry_mgr->CounterFamily(
+				"zeek", "sessions", {"protocol"},
+				"Total number of sessions", "1", true);
+
+			auto [it, inserted] = entries.insert(
+				{protocol, Block{num_family, total_family, protocol}});
+
+			if ( inserted )
+				return it;
+
+			return entries.end();
+			}
+
+		Block* GetCounters(std::string protocol)
```

Comment:
This copies the string every time we call this function. Can't we use a `string_view` here instead? Or some enum instead (which could allow us to use the enum as an index directly).

Reply:
I changed the argument for this and `InitCounters` to be a `const std::string&` since the map stores that type as the key. I can't use an enum since this is eventually possibly used for other plugins, and we wouldn't want them to have to modify the enum for it.